test/plugins/lock.test.js

```diff
@@ -60,9 +47,14 @@ describe('lock plugin test', () => {
         it('explicitly disable', () => {
             process.env.REDLOCK_ENABLED = 'false';
 
-            /* eslint-disable global-require */
-            const plugin = require('../../plugins/lock');
-            /* eslint-enable global-require */
+            const testConfig = importFresh('config');
+            /* eslint-disable prettier/prettier */
+            const plugin = rewiremock.proxy('../../plugins/lock', {
+                'config': testConfig,
+                'ioredis': RedisMock,
+                'redlock': RedLockMock
+            });
+            /* eslint-enable prettier/prettier */
```

Comment:
any chance we can reduce redundancy by putting this in a beforeEach?

Reply:
it has to be this way because we need to override process env before loading `config`, that why we are using `importFresh` here. Each test cases use different process env so we can't really do it in beforeEach